From 5c12367abd3e95def5df02437a0fc2bedf500119 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Mon, 21 May 2018 13:30:49 +0300
Subject: [PATCH 3832/5242] MLK-17531-4: ARM64: dts: mx8mm-evk: enable AK4497
 codec

commit  afb54cead3f0b99dcf8882997c4a27fad1499fe2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable AK4497 with mode 0. For ak4497 the same SAI interface as
for AK4458 is used, so a separate ak4497 dts is needed.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts   |   96 ++++++++++++++++++++
 2 files changed, 98 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 200c112..696cfb2 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -76,4 +76,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-evk-ak4497.dtb \
 				 fsl-imx8mq-evk-audio-tdm.dtb \
 				 fsl-imx8mq-evk-drm.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb \
+				 fsl-imx8mm-evk-ak4497.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts
new file mode 100644
index 0000000..3e07f54
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak4497.dts
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mm-evk.dts"
+
+/ {
+	sound-ak4458 {
+		status = "disabled";
+	};
+
+	sound-ak4497 {
+		status = "okay";
+	};
+};
+
+&iomuxc {
+	imx8mm-evk {
+		pinctrl_sai1_pcm: sai1grp_pcm {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
+				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
+				MX8MM_IOMUXC_SAI1_RXD7_SAI1_TX_SYNC	0xd6
+				MX8MM_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
+				MX8MM_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
+				MX8MM_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1	0xd6
+				MX8MM_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2	0xd6
+				MX8MM_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3	0xd6
+				MX8MM_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4	0xd6
+				MX8MM_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5	0xd6
+				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
+				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
+			>;
+		};
+
+		pinctrl_sai1_dsd: sai1grp_dsd {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
+				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
+				MX8MM_IOMUXC_SAI1_RXD7_SAI1_TX_DATA4	0xd6
+				MX8MM_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
+				MX8MM_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
+				MX8MM_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1	0xd6
+				MX8MM_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2	0xd6
+				MX8MM_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3	0xd6
+				MX8MM_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4	0xd6
+				MX8MM_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5	0xd6
+				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
+				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
+			>;
+		};
+
+		pinctrl_sai1_dsd512: sai1grp_dsd512 {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI1_MCLK_SAI1_TX_BCLK	0xd6
+				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
+				MX8MM_IOMUXC_SAI1_RXD7_SAI1_TX_DATA4	0xd6
+				MX8MM_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
+				MX8MM_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
+				MX8MM_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1	0xd6
+				MX8MM_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2	0xd6
+				MX8MM_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3	0xd6
+				MX8MM_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4	0xd6
+				MX8MM_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5	0xd6
+				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
+				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
+			>;
+		};
+	};
+};
+
+&sai1 {
+	pinctrl-names = "default", "dsd", "dsd512";
+	pinctrl-0 = <&pinctrl_sai1_pcm>;
+	pinctrl-1 = <&pinctrl_sai1_dsd>;
+	pinctrl-2 = <&pinctrl_sai1_dsd512>;
+	assigned-clocks = <&clk IMX8MM_CLK_SAI1_SRC>,
+			<&clk IMX8MM_CLK_SAI1_DIV>;
+	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL2_OUT>;
+	assigned-clock-rates = <0>, <45158400>;
+	fsl,sai-multi-lane;
+	fsl,dataline,dsd = <0xff 0x11>;
+	dmas = <&sdma2 0 26 0>, <&sdma2 1 26 0>;
+	status = "okay";
+};
+
-- 
1.7.9.5

